### DISCUSSION TITLE
How to rewrap interferogram for better visualization?

### DISCUSSION THREAD

**Discussion by leeyhgeo** [isce-framework/isce2#657] [Q&A] [answered]
I'm recently processing DInSAR for a specific earthquake event. However, I ran into a problem that the surface displacement is too large to clearly visualize fringes (all fringes stick together and look like noises). I found that some researchers might rewrap the interferogram to better visualize it (e.g., from 5 cm/cycle to 60 cm/cycle). Is it possible to do so by ISCE or other common software?

**Comment by y-okur** [comment, marked as answer]
Hello,

Assuming that there are no problems with unwrapping, for visualization, you can do something like this in python:

`data = np.subtract(np.mod(data,2*np.pi), np.pi)`

to rewrap the phase with pi cycles. Here data is the unwrapped phase from ISCE outputs (exp: filt_******-******_**rlks_**alks.unw.geo). In your case pi cycle will be ~5cm, and 2pi will be ~10cm. For 60 cm you need to use 12pi cycle. So the code will be:

`data = np.subtract(np.mod(data,12*np.pi), 6*np.pi)`

Using C-band with large earthquakes, it is not very easy to unwrap the phase because the deformation near the fault will be very large and there will be unwrapping errors.

I hope this helps.

**Reply by leeyhgeo** [reply]
Thank you for the clear explanation!
I followed your steps and it successfully worked.

**Comment by EJFielding** [comment]
For large earthquakes, with surface displacments of many meters, the fringes will be lost (stick or merge together) close to the faults. Once the fringes are lost due to large displacements, the phase will look like noise and cannot be unwrapped. You should use the dense offsets method to measure large displacements.

**Comment by fdelgadodelapuente** [comment]
Yes, you can. Open the interferogram with mdx.py, right ciick button on unw, then change wrap to whatever phase you want in radians. So if the natural fringe rate of a Sentinel-1 interferogram is 2pi rads per 2.77 cm, then 20 rads will show one fringe per every 9 cm approximately.  This assumes that the interferogram is correctly unwrapped which Is not the case for large strain due to surface ruptures.
